### PR TITLE
Add particle position shift.

### PR DESCRIPTION
**Pull request by abhinavmuta**
This is added to maintain better uniform
distribution of particles.

### CODE REVIEW

File: pysph/sph/wc/shift.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+class SimpleShift(Equation):
+    """See the paper by R. Xu et al. (JCP 2009), equation(35)
+    """
+    def __init__(self, dest, sources, const=0.04):
+        self.beta = const
+        super(SimpleShift, self).__init__(dest, sources)
+
+    def py_initialize(self, dst, t, dt):
+        from math import sqrt
```
Comment (prabhuramachandran) [inline pysph/sph/wc/shift.py]: Is this import needed? and shouldn't it be from `numpy import sqrt`?

File: pysph/sph/wc/shift.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        vmag = dst.u[:]*dst.u[:] + dst.v[:]*dst.v[:] + dst.w[:]*dst.w[:]
```
Comment (prabhuramachandran) [inline pysph/sph/wc/shift.py]: Can't this be simply `vmag = dst.u**2 + dst.v **2 + dst.w**2`

File: pysph/sph/wc/shift.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        vmax = serial_reduce_array(vmag, 'max')
```
Comment (prabhuramachandran) [inline pysph/sph/wc/shift.py]: Technically this should be a `parallel_reduce_array` but it is fine for this use case I imagine.

File: pysph/sph/wc/shift.py [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+        d_dpos[d_idx*3] = fac*grad_c[0]
+        d_dpos[d_idx*3 + 1] = fac*grad_c[1]
+        d_dpos[d_idx*3 + 2] = fac*grad_c[2]
+
+        d_x[d_idx] += d_dpos[d_idx*3]
+        d_y[d_idx] += d_dpos[d_idx*3 + 1]
+        d_z[d_idx] += d_dpos[d_idx*3 + 2]
+
+
+class CorrectVelocities(Equation):
```
Comment (prabhuramachandran) [inline pysph/sph/wc/shift.py]: Docstring for this?

File: pysph/sph/wc/shift.py [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+        d_x[d_idx] += d_dpos[d_idx*3]
+        d_y[d_idx] += d_dpos[d_idx*3 + 1]
+        d_z[d_idx] += d_dpos[d_idx*3 + 2]
+
+
+class FickianShift(Equation):
+    """See the paper by S. J. Lind et al. (JCP 2012), equation(21-24),
+    for the constant see A. Skillen et al. (CMAME 2013), equation(13).
+    """
+    def __init__(self, dest, sources, const=10, tensile_const=0.2,
```
Comment (prabhuramachandran) [inline pysph/sph/wc/shift.py]: Why the name `const`, can there be a better name that is similar to the one used in the paper?

File: pysph/sph/wc/shift.py (hunk 2 above)
Comment (abhinavmuta) [reply pysph/sph/wc/shift.py]: fixed, thanks!

File: pysph/sph/wc/shift.py (hunk 3 above)
Comment (abhinavmuta) [reply pysph/sph/wc/shift.py]: changed it, thanks.

File: pysph/sph/wc/shift.py (hunk 4 above)
Comment (abhinavmuta) [reply pysph/sph/wc/shift.py]: added doc strings.

File: pysph/sph/wc/shift.py (hunk 5 above)
Comment (abhinavmuta) [reply pysph/sph/wc/shift.py]: the papers names it as 'C', now changed it to 'fickian_const', should I change it ?

File: pysph/sph/wc/shift.py (hunk 1 above)
Comment (abhinavmuta) [reply pysph/sph/wc/shift.py]: this is not needed! I tried to not import numpy at all, but now changed it to a numpy.sqrt operation as it is much clear.

**Comment by prabhuramachandran** [comment]
LGTM, thanks!

**Pull request merged**